src/controllers/handlers/update-member-role-handler.ts

```diff
@@ -0,0 +1,63 @@
+import { HandlerContextWithPath, HTTPResponse } from '../../types'
+import { InvalidRequestError, NotAuthorizedError } from '@dcl/platform-server-commons'
+import { CommunityNotFoundError } from '../../logic/community'
+import { errorMessageOrDefault } from '../../utils/errors'
+import { EthAddress } from '@dcl/schemas'
+import { CommunityRole } from '../../types/entities'
+
+export async function updateMemberRoleHandler(
+  context: Pick<
+    HandlerContextWithPath<'community' | 'logs', '/v1/communities/:id/members/:address'>,
+    'components' | 'params' | 'verification' | 'url'
+  >
+): Promise<HTTPResponse> {
+  const {
+    components: { community, logs },
+    params: { id: communityId, address: targetAddress },
+    verification,
+    url
+  } = context
+
+  const logger = logs.getLogger('update-member-role-handler')
+
+  try {
+    const updaterAddress = verification!.auth.toLowerCase()
+    const targetAddressLower = targetAddress.toLowerCase()
+
+    if (!EthAddress.validate(targetAddressLower)) {
+      throw new InvalidRequestError(`Invalid address provided`)
+    }
+
+    const searchParams = new URLSearchParams(url.search)
+    const newRole = searchParams.get('role')
+
+    if (!newRole || !Object.values(CommunityRole).includes(newRole as CommunityRole)) {
```

Comment:
Can we move this check within `community.updateMemberRole` so every piece of code that calls it gets this validation?